ptt.php

```diff
@@ -1,156 +1,179 @@
-    foreach($phones as $phone) {
-        push2phone($phone, "RTPRx:Stop,RTPTx:Stop,SoftKey:Exit", $user, $pwd);
-        // If this is one way PTT (phone sending audio is not receiving it - it is not listed in phones array)
-        if(!in_array($_SERVER['REMOTE_ADDR'], $phones)) {
-            // Send text to phone advising PTT ended, otherwise phone will complain about XML format
-            echo "<CiscoIPPhoneText><Title>PTT</Title><Text>PTT session ended.</Text></CiscoIPPhoneText>";
-        }
-    }
-    die();
-} else {						// if PTT session is still in progress
-    // Check for istalking, if it doesn't exist this must be
-    // the first time the phone has called this page
-    if(isset($_GET['istalking'])) {
-        $istalking = $_GET['istalking'];
-    }
-    else {
-        // Must be our initial request for the page, lets tell everyone to listen
-        $istalking = "false";
-        foreach($phones as $phone) {
-            push2phone($phone, "RTPMRx:{$MCAddress}:{$MCPort}", $user, $pwd);
-        }
-    }
-}
+	function __construct($deviceIp, $authName = 'cisco', $authPassword = 'cisco'){
+		$this->ip 			= $deviceIp;
+		$this->authName 	= $authName;
+		$this->authPassword = $authPassword;
+	}
+
+
+	function push($xml){
+		$response = '';
+
+		$auth = base64_encode($this->authName.':'.$this->authPassword);
+		$postData = "XML=".urlencode($xml);
 
-// Set phone's app window title and promt
-echo "<CiscoIPPhoneText><Title>PTT</Title><Prompt>Use soft keys to talk</Prompt>";
-
-// PTT operation
-if($istalking != "true") {
-    echo "<Text>Press and hold the [Talk] soft key to transmit. If you want to speak continuously without holding a button down, you can press and release the [Lock] soft key.</Text>";
-
-    echo "<SoftKeyItem>";
-    echo "<Name>Talk</Name>";
-    echo "<URL>RTPTx:Stop</URL>";
-    echo "<Position>1</Position>";
-    echo "<URLDown>RTPMTx:{$MCAddress}:{$MCPort}</URLDown>";
-    echo "</SoftKeyItem>";
-
-    echo "<SoftKeyItem>";
-    echo "<Name>Exit</Name>";
-    echo "<URL>{$URLBase}ptt.php?name={$MAC}&amp;close=true</URL>";
-    echo "<Position>3</Position>";
-    echo "<URLDown>RTPRx:Stop</URLDown>";
-    echo "</SoftKeyItem>";
-
-    echo "<SoftKeyItem>";
-    echo "<Name>Lock</Name>";
-    echo "<URL>{$URLBase}ptt.php?name={$MAC}&amp;istalking=true</URL>";
-    echo "<Position>4</Position>";
-    echo "<URLDown>RTPMTx:{$MCAddress}:{$MCPort}</URLDown>";
-    echo "</SoftKeyItem>";
+		$post = "POST /CGI/Execute HTTP/1.0\r\n";
+		$post .= "Host: {$this->ip}\r\n";
+		$post .= "Authorization: Basic {$auth}\r\n";
+		$post .= "Connection: close\r\n";
+		$post .= "Content-Type: application/x-www-form-urlencoded\r\n";
+		$post .= "Content-Length: ".strlen($postData)."\r\n\r\n";
+		$post .= $postData;
+
+		$fp = @fsockopen ($this->ip, 80, $errno, $response, 10);
+		if($fp){
+			fputs($fp, $post);
+			flush();
+			while (!feof($fp)) {
+				$response .= fgets($fp, 1024);
+				flush();
+			}
+
+		}
+
```

Comment:
error_log("pushed:" . $xml, 0);
                error_log("to:" . $this->ip, 0);
                error_log("response:" . $response, 0);

                // we should check the return value !